src/gt4py/next/iterator/embedded.py

```diff
@@ -1603,6 +1603,14 @@ def set_at(expr: common.Field, domain: common.DomainLike, target: common.Mutable
     operators._tuple_assign_field(target, expr, common.domain(domain))
 
 
+@runtime.if_stmt.register(EMBEDDED)
+def if_stmt(cond: bool, true_branch: Callable[[], None], false_branch: Callable[[], None]) -> None:
```

Comment:
Let's document somewhere why we have to represent the branches as callables (or more general how we represent `stmt`s in expression trees.

Reply:
I've added a docstring here.

| or more general how we represent stmts in expression trees

Makes sense, but I skipped this for now.